name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -1184,16 +1184,53 @@ private void addDividendMarkerLines(ChartInterval chartInterval)
                     IAxis xAxis = chart.getAxisSet().getXAxis(0);
                     IAxis yAxis = chart.getAxisSet().getYAxis(0);
 
+                    int yPosLabel = 0;
+                    int lastWriteLabelLevel1 = 0;
+                    int lastWriteLabelLevel2 = 0;
+                    int lastWriteLabelLevel3 = 0;
+
                     for (int index = 0; index < dates.length; index++)
                     {
+                        boolean freeSpaceForLabelLevel1 = true;
+                        boolean freeSpaceForLabelLevel2 = true;
+                        boolean freeSpaceForLabelLevel3 = true;
+
                         int x = xAxis.getPixelCoordinate(dates[index].getTime());
                         int y = yAxis.getPixelCoordinate(values[index]);
 
                         String label = getDividendLabel(dividends.get(index));
                         Point textExtent = event.gc.textExtent(label);
 
                         event.gc.setForeground(Colors.theme().defaultForeground());
-                        event.gc.drawText(label, x - (textExtent.x / 2), y - 22, true);
+
+                        if (((x - (textExtent.x / 2)) - lastWriteLabelLevel1) <= 0)
+                            freeSpaceForLabelLevel1 = false;
```

Comment:
Why not using a ternary operator instead of doing an assignement to `true`and maybe switching it to false into a for loop?